bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java

```diff
@@ -267,6 +274,42 @@ public void getURLInput(String url) throws IOException {
         }
     }
 
+    /**
+     * Builds the {@link TimeSeries} that represents the future tibber prices.
+     *
+     * @param today The prices for today
+     * @param tomorrow The prices for tomorrow.
+     * @return The {@link TimeSeries} with future values.
+     */
+    private TimeSeries buildTimeSeries(JsonArray today, JsonArray tomorrow) {
+        final TimeSeries timeSeries = new TimeSeries(TimeSeries.Policy.REPLACE);
+        mapTimeSeriesEntries(today, timeSeries);
+        mapTimeSeriesEntries(tomorrow, timeSeries);
+        return timeSeries;
+    }
+
+    private void mapTimeSeriesEntries(JsonArray prices, TimeSeries timeSeries) {
+        for (JsonElement entry : prices) {
+            final Instant startsAt = parseDateTime(entry.getAsJsonObject().get("startsAt"));
+            if (startsAt == null) {
+                // Skip entry, if date is invalid.
+                continue;
+            }
+            final DecimalType value = new DecimalType(entry.getAsJsonObject().get("total").getAsString());
+            timeSeries.add(startsAt, value);
+        }
+    }
+
+    @Nullable
+    private Instant parseDateTime(JsonElement element) {
+        try {
+            return ZonedDateTime.parse(element.getAsString()).toInstant();
+        } catch (DateTimeParseException e) {
+            logger.warn("Could not parse timestamp from response.", e);
```

Comment:
This will log a warning per entry. As it is most likely that all entries will be either valid or invalid, not just some of them, I would suggest to catch this exception in the `getURLInput` method. You could even add the exception in 213 and handle it the same way as `JsonSyntaxException` already caught there. That will exit the loop fast and also consolidate the error handling while also setting the thing status to offline.

Consider also if you need this private method at all, since it's only called from line 293, and it's a one-liner. Perhaps ZonedDateTime.parse could be called directly from `mapTimeSeriesEntries`.